### PR TITLE
feat(backend): Calculate circulating supply

### PR DESCRIPTION
**Pull request by telezhnaya**
Closes (in one day) #489

### CODE REVIEW

**Comment by telezhnaya** [comment]
It's too early to review that.
I feel that I will need to give links and ask questions soon, that's why I open it. I'll ping you directly when it's ready for review.

File: backend/src/db-utils.js [hunk 1]
```diff
@@ -529,6 +529,24 @@ const queryPartnerUniqueUserAmount = async () => {
...
 
+const getLockupAccountIds = async (blockHeight) => {
+  return await queryRows(
+    [
+      `SELECT accounts.account_id
+         FROM accounts
+                LEFT JOIN receipts AS receipts_start ON accounts.created_by_receipt_id = receipts_start.receipt_id
+                LEFT JOIN blocks AS blocks_start ON receipts_start.included_in_block_hash = blocks_start.block_hash
+                LEFT JOIN receipts AS repecipts_end ON accounts.deleted_by_receipt_id = repecipts_end.receipt_id
+                LEFT JOIN blocks AS blocks_end ON repecipts_end.included_in_block_hash = blocks_end.block_hash
```
Comment (frol) [inline backend/src/db-utils.js:540]:
```suggestion
                LEFT JOIN receipts AS receipts_end ON accounts.deleted_by_receipt_id = receipts_end.receipt_id
                LEFT JOIN blocks AS blocks_end ON receipts_end.included_in_block_hash = blocks_end.block_hash
```

File: backend/src/index.js [hunk 2]
```diff
@@ -230,6 +236,26 @@ function startStatsAggregation() {
   setTimeout(regularStatsAggregate, 0);
 }
 
+function startRegularCalculationCirculatingSupply() {
+  const regularCalculateCirculatingSupply = async () => {
+    console.log(`Starting regular calculation of circulating supply...`);
+    try {
+      // TODO put here needed blockHeight
+      await calculateCirculatingSupply(37285191);
```
Comment (frol) [inline backend/src/index.js:244]: TODO reminder

**Comment by frol** [comment]
Action item or @frol:

* [ ] add wamp endpoint exposing the circulating supply value

**Comment by telezhnaya** [comment]
> Action item for @frol:
> 
> * [ ]  add wamp endpoint exposing the circulating supply value

I didn't want to commit anything from `frontend`, please check out the last commit, maybe it was done before
(I have no idea whether it's correct or not)

File: backend/src/db-utils.js [hunk 3]
```diff
@@ -529,6 +529,38 @@ const queryPartnerUniqueUserAmount = async () => {
...
+
+const getLastYesterdayBlock = async (timestamp) => {
+  return await querySingleRow(
+    [
+      // we can't cast timestamp to number (it's too big), BN is not OK here, that's why I concatenate strings
+      `SELECT block_height
+       FROM blocks
+       WHERE block_timestamp < ` +
+        timestamp.toString() +
+        `\nORDER BY block_timestamp DESC LIMIT 1;`,
```
Comment (frol) [inline backend/src/db-utils.js:558]: This should work, I believe:

```suggestion
       WHERE block_timestamp < :blockTimestamp
       ORDER BY block_timestamp DESC LIMIT 1;`,
       { blockTimestamp: timestamp.toString() },
```

Comment (telezhnaya) [reply backend/src/db-utils.js:558]: Nope, there are quotes 😞

**Review by frol — APPROVED** [review]

**Pull request merged**